filters.go

```diff
@@ -35,9 +36,22 @@ func (filterPredicate *FilterPredicate) Validate() error {
 		PredicateTypeEnumEndsWith,
 		PredicateTypeEnumStartsWith,
 	}
-	if filterPredicate.CaseSensitive != nil && !slices.Contains(caseSensitiveTypes, filterPredicate.Type) {
+	if filterPredicate.CaseSensitive != nil &&
```

Comment:
Why is `*filterPredicate.CaseSensitive` necessary? I think the predicate would still be incorrect if if case sensitive was set to false.

Reply:
This is true. I'm testing out some Terraform workflows and this was a workaround that I don't think we need